modules/transactions/src/main/java/org/apache/ignite/internal/tx/HybridTimestampTrackerImpl.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.tx;
+
+import static org.apache.ignite.internal.hlc.HybridTimestamp.NULL_HYBRID_TIMESTAMP;
+
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.ignite.internal.hlc.HybridTimestamp;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Hybrid timestamp tracker.
+ */
+public class HybridTimestampTrackerImpl implements HybridTimestampTracker {
```

Comment:
You don't need this change. It brings a lot of modified files in PR without any profit.
Instead, change implementation in the following way:
```
public class HybridTimestampTracker {
    /** Timestamp. */
    private final AtomicLong timestamp = new AtomicLong(NULL_HYBRID_TIMESTAMP);
    
    private final Consumer<HybridTimestamp> updateTs;

    public HybridTimestampTracker() {
        this.updateTs = timestamp -> {
            long tsVal = HybridTimestamp.hybridTimestampToLong(ts);

            HybridTimestampTracker.this.timestamp.updateAndGet(x -> Math.max(x, tsVal));
        };
    }
    
    public HybridTimestampTracker(@Nullable HybridTimestamp currentTs, Consumer<HybridTimestamp> updateTs) {
        this.timestamp.set(currentTs.longValue());
        this.updateTs = updateTs;
    }

    public @Nullable HybridTimestamp get() {
        return HybridTimestamp.nullableHybridTimestamp(timestamp.get());
    }

    public void update(@Nullable HybridTimestamp ts) {
        updateTs.accept(ts);
    }
}
```

Reply:
I re-wrote HybridTimestampTracker, but in a different approach.